src/Tools/Source/FakeSign/Program.cs

```diff
@@ -81,9 +81,12 @@ private static bool Validate(PEReader peReader)
             }
 
             CorHeader header = peReader.PEHeaders.CorHeader;
-            if ((header.Flags & CorFlags.StrongNameSigned) == CorFlags.StrongNameSigned)
+            var expectedStrongNameFlag = unSign ? CorFlags.StrongNameSigned : 0;
+            var actualStrongNameFlag = header.Flags & CorFlags.StrongNameSigned;
+
+            if (expectedStrongNameFlag != actualStrongNameFlag)
             {
-                Console.Error.WriteLine("PE file is already strong-name signed.");
+                Console.Error.WriteLine($"PE file is {(unSign ? "not" : "already")} strong-name signed.");
                 return false;
```

Comment:
I addressed that issue (FakeSign running on a binary that hadn't been rebuilt) in #309
If you are seeing this in your builds, then you need to look at why incremental build is failing (probably do a 'scorch' or force a rebuild with /t:Rebuild).

You could add "-f" if you feel it's necessary, but I think the default is adding some value to diagnose incremental build errors. Note that "-f" touches the file, so it's not a complete "no-op".